noodles-editor/src/noodles/utils/node-creation-utils.ts

```diff
@@ -0,0 +1,151 @@
+import type { NodeJSON } from '@xyflow/react'
+import { type MathOpType, mathOps, type OpType, opTypes } from '../operators'
+import { edgeId, nodeId } from './id-utils'
+
+export type NodeType = OpType | MathOpType | 'ForLoop'
+
+
+// Get all available node types (operators, math ops, and special types like ForLoop)
+export function getNodeTypeOptions(): NodeType[] {
+  return (Object.keys(opTypes) as NodeType[])
+    .filter(type => type !== 'ForLoopBeginOp' && type !== 'ForLoopEndOp')
+    .concat(['ForLoop', ...Object.keys(mathOps)])
+    .sort()
+}
+
+// Convert a string to kebab-case
+function toKebabCase(str: string): string {
+  return str.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase()
+}
+
+// Generate an operator ID from its type
+function makeOpId(type: OpType, containerId: string): string {
+  const baseName = toKebabCase(type.replace(/Op$/g, ''))
+  return nodeId(baseName, containerId)
+}
```

Comment:
Are these only used in this module? They seem like something we'd use elsewhere too but I'm not sure off the top of my head